litgpt/pretrain.py

```diff
@@ -104,22 +133,33 @@ def setup(
     out_dir = init_out_dir(out_dir)
     # in case the dataset requires the Tokenizer
     tokenizer = Tokenizer(tokenizer_dir) if tokenizer_dir is not None else None
-
+    
     logger = choose_logger(
         logger_name, out_dir, name=f"pretrain-{config.name}", resume=resume, log_interval=train.log_interval
     )
 
     if devices > 1:
-        strategy = FSDPStrategy(auto_wrap_policy={Block}, state_dict_type="full", sharding_strategy="HYBRID_SHARD")
+        if isinstance(replication_size, int) ^ isinstance(sharding_size, int):
+            raise ValueError("Either `replication_size` and `sharding_size` must both be set or left undefined")
+        device_mesh = (replication_size, sharding_size) if replication_size else None
+        strategy = FSDPStrategy(auto_wrap_policy={Block}, state_dict_type="full", sharding_strategy="HYBRID_SHARD", device_mesh=device_mesh)
     else:
         strategy = "auto"
-    fabric = L.Fabric(devices=devices, strategy=strategy, precision=precision, loggers=[logger])
+    fabric = L.Fabric(devices=devices, num_nodes=num_nodes, strategy=strategy, precision=precision, loggers=[logger])
     fabric.launch()
 
     fabric.print(pprint.pformat(hparams))
     if logger_name in ("tensorboard", "wandb"):
         fabric.logger.log_hyperparams(hparams)
 
+    # Configure profiler
+    use_pt_profiler=False
```

Comment:
Can you make this configurable from the command line so that we can enable/disable it without needing to rebuild the docker image?

Reply:
Done.